sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -309,9 +309,7 @@ static gnrc_rpl_parent_t *_gnrc_rpl_find_preferred_parent(gnrc_rpl_dodag_t *doda
         return NULL;
     }
 
-    LL_FOREACH(dodag->parents, elt) {
-        new_best = dodag->instance->of->which_parent(new_best, elt);
-    }
+    LL_SORT(dodag->parents, dodag->instance->of->which_parent);
```

Comment:
Ah, you're right there, that wasn't supposed to happen. An `new_best = dodag->parents` after the `LL_SORT` should be sufficient to restore the old behaviour